pkg/system/filesys_windows.go

```diff
@@ -32,7 +32,7 @@ func MkdirAllWithACL(path string, perm os.FileMode, sddl string) error {
 
 // MkdirAll implementation that is volume path aware for Windows.
 func MkdirAll(path string, _ os.FileMode, sddl string) error {
-	return mkdirall(path, false, sddl)
+	return mkdirall(path, sddl != "", sddl)
```

Comment:
This change kind of obsoletes `MkdirAllWithACL` since it applies based on the value of `sddl`. This approach seems fine, but it might be better to defer to a later time, with the removal of `MkdirAllWithACL`, if that makes sense. It would be a good idea to review why `MkdirAllWithACL` was added in the first place.